d2ir/compile.go

```diff
@@ -255,6 +274,21 @@ func (c *compiler) compileMap(dst *Map, ast, scopeAST *d2ast.Map) {
 				ScopeMap: dst,
 				ScopeAST: scopeAST,
 			})
+		case n.Substitution != nil:
+			if !n.Substitution.Spread {
+				c.errorf(n.Import, "invalid non-spread substitution in map")
```

Comment:
I could be wrong but I believe the parser will already error on this? If not, do add a test.

Reply:
it doesn't, but ur right that it doesn't reach here. added a comment that it should

Reply:
Why add a comment? Why doesn't it reach here?

Reply:
did you read the comment? it doesn't error but it's quite harmless.

Reply:
Oh I see. I would still say it's unnecessary. If that behavior ever changed, it would be on the parser to error, not the IR as it's invalid syntax.

Reply:
But I'll defer.